src/main/java/ru/test/restservice/service/ProjectService.java

```diff
@@ -50,44 +51,46 @@ public void createNewProject(String projectName, Authentication auth) throws IOE
     }
 
     public void deleteProject(UUID projectId) throws IOException {
-        Project projectToDelete = projectRepository.findById(projectId);
+        Project projectToDelete = projectRepository.findById(projectId).get();
         FileSystemUtils.deleteRecursively(Paths.get(projectToDelete.path));
         projectRepository.delete(projectToDelete);
     }
 
-    public ModelAndView openProject(UUID projectId, Authentication auth) {
-        Project project = projectRepository.findById(projectId);
-        User user = userRepository.findByEmail(auth.getName());
-        if (project != null && project.owners.contains(user)) {
-            ModelAndView main = new ModelAndView("index");
-            main.addObject("projectId", projectId);
-            return main;
-        } else {
+    public void checkOwnership(UUID projectId, String email) {
+        Optional<Project> project = projectRepository.findById(projectId);
```

Comment:
```suggestion
        Project project = projectRepository.findById(projectId)
             .orElseThrow(NotFoundException::new)
```
можно вот так сделать
и для user тоже